docs/Technical-Documentation/jwt-key-rotation.md

```diff
@@ -0,0 +1,70 @@
+# Rotating JWT Keys
+
+## Context
+
+To maintain good security, we will periodically rotate our JWT keys. This document outlines the process of how to do this.
+
+**Warning** Production sites will need to be taken down for maintenance when rotating keys, as the rotation will automatically invalidate all current sessions.
+
+### 1. Generate New Keys
+
+In your Mac terminal (or bash terminal in Windows), enter the following command:
+```bash=
+ssh-keygen -t rsa -b 4096 -m PEM -f jwtRS256.key
+```
+You will receive the following response. Do not add a passphrase when prompted.
+```
+Enter passphrase (empty for no passphrase):
+Enter same passphrase again:
+Your identification has been saved in jwtRS256.key
+Your public key has been saved in jwtRS256.key.pub
+The key fingerprint is:
+# returns key fingerprint
+The key's randmart image is:
+# returns key's randmart image
+```
+Then enter
+```bash=
+openssl rsa -in jwtRS256.key -pubout -outform PEM -out jwtRS256.key.pub
+```
+
+You can now check the contents of your keys with these commands
+```bash=
+cat jwtRS256.key
+# returns private key
+cat jwtRS256.key.pub
+# returns public key
+```
+
+### 2. Base64 Encode Private Key
+
+We use Base64 Encoded Private Keys to make it easier to save to cloud environments and local `.env` files.
+
+```bash=
+openssl enc -base64 -in jwtRS256.key -out jwtRS256.key.base64
+cat jwtRS256.key.base64
+```
+
+### 3. Copy Keys
+
+#### Dev/Staging Environments
+1. Distribute the private key to development staff securely to copy to `.env` files as the value for key `JWT_KEY`
+2. Update the variable `JWT_KEY` with the private key in cloud.gov backend development and staging environments
+3. Login to the [Login.gov Sandbox](https://dashboard.int.identitysandbox.gov/) and update the public key there
+
+#### CI/CD Environment
+**Note** _Please generate a separate set of keys for the CI/CD environment_
+1. Distribute the private key to development staff securely to copy to `.env` files as the value for key `JWT_CERT_TEST`
+2. Update the variable `JWT_KEY_TEST` in CircleCI with the new public key.
```

Comment:
I see we have both `JWT_CERT_TEST` and `JWT_KEY_TEST` saved in Circle CI currently but I don't see a reference to `JWT_KEY_TEST` anywhere in the code base - are we supposed to have both?

Reply:
I think we need `JWT_KEY_TEST` in CircleCI and `JWT_CERT_TEST` defined in our local `.env` files to send to CircleCI. I did notice we have some things defined in different places that we don't need, so a follow up issue to clean those up would probably be in order.

Reply:
Looks like we have a stub ticket we can use to track this: https://github.com/raft-tech/TANF-app/issues/597